Pokemon/Pokemon/Models/Pokemon.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  Pokemon.swift
+//  Pokemon
+//
+//  Created by Cora Jacobson on 7/18/20.
+//  Copyright © 2020 Cora Jacobson. All rights reserved.
+//
+
+import Foundation
+
+struct Pokemon: Codable {
+    
+    enum Keys: String, CodingKey {
+        case name
+        case id
+        case types
+        case abilities
+        case imageURL = "sprites"
+        
+        enum TypeDescriptionKeys: String, CodingKey {
+            case type
+            
+            enum TypeKeys: String, CodingKey {
+                case name
+            }
+        }
+        
+        enum AbilityDescriptionKeys: String, CodingKey {
+            case ability
+            
+            enum AbilityKeys: String, CodingKey {
+                case name
+            }
+        }
+        
+        enum SpritesKey: String, CodingKey {
+            case front_default
+        }
+        
+    }
+    
+    var name: String
+    var id: Int
+    var types: [String]
+    var abilities: [String]
+    var imageString: String
+    
+    init(_ name: String, _ id: Int, _ types: [String], _ abilities: [String], _ imageString: String) {
+        self.name = name
+        self.id = id
+        self.types = types
+        self.abilities = abilities
+        self.imageString = imageString
+    }
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: Keys.self)
+        name = try container.decode(String.self, forKey: .name)
```

Comment:
Add in conditional code to decode the flat version if you had set that userInfo key on the decoder:

``` swift
if let flatData = decoder.userInfo[.flatData] as? Bool, flatData == true {
    self.name = try container.decode(String.self, forKey: .name)
    self.id = try container.decode(Int.self, forKey: .id)
    self.types = try container.decode([String].self, forKey: .types)
    self.abilities = try container.decode([String].self, forKey: .abilities)
    self.imageString = try container.decode(String.self, forKey: .imageString)
    
    return
}
```